dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceCache.java

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+        ignoreGlobalVM = ImmutableSet.copyOf(holder);
+    }
+
+    public String[] getMacro(String name) {
+
+
+        String[] rw = null;
+        try {
+            rw = (String[]) cache.get(MACRO_PREFIX + name, macroCacheGroup);
+        } catch (DotCacheException e) {
+            Logger.debug(this, "Cache Entry not found", e);
+        }
+        return rw;
+
+    }
+
+    public void putMacro(String name, String content) {
+        if (name == null || content == null) {
+            Logger.warn(this.getClass(), "Cannot add a null macro to cache:" + name + " / " + content);
+            return;
+        }
+        String[] rw = {name, content};
+        cache.put(MACRO_PREFIX + name, rw, macroCacheGroup);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#get(java.lang.Object)
+     */
+    public Resource get(Object resourceKey) {
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+        Resource resource = null;
+        String key = group + cleanedResourceKey;
+        try {
+            resource = (Resource) cache.get(key, group);
+        } catch (DotCacheException e) {
+            Logger.debug(this, "Cache Entry not found", e);
+        }
+        return resource != null ? resource : null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.apache.velocity.runtime.resource.ResourceCache#initialize(org.apache.velocity.runtime.
+     * RuntimeServices)
+     */
+    public void initialize(RuntimeServices rs) {
+        cache = CacheLocator.getCacheAdministrator();
+    }
+
+    public void addMiss(Object resourceKey) {
+        Logger.info(this.getClass(), "velocityMiss:" + resourceKey);
+    }
+
+    public boolean isMiss(Object resourceKey) {
+        return false;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#put(java.lang.Object,
+     * org.apache.velocity.runtime.resource.Resource)
+     */
+    public Resource put(Object resourceKey, Resource resource) {
+        if (resource != null && ignoreGlobalVM.contains(resource.getName())) {
+            return resource;
+        }
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+
+
+        String key = group + cleanedResourceKey;
+        // Add the key to the cache
+        cache.put(key, resource, group);
+
+        return resource;
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#remove(java.lang.Object)
+     */
+    public Resource remove(Object resourceKey) {
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+
+        String key = group + cleanedResourceKey;
+
+        try {
+            cache.remove(key, group);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)